OpenSim/Common/C3DFileAdapter.cpp

```diff
@@ -108,13 +112,22 @@ C3DFileAdapter::extendRead(const std::string& fileName) const {
 
         double time_step{1.0 / acquisition->GetPointFrequency()};
         for(int f = 0; f < marker_nrow; ++f) {
-            SimTK::RowVector_<SimTK::Vec3> row{marker_pts->GetItemNumber()};
+            SimTK::RowVector_<SimTK::Vec3> row{ marker_pts->GetItemNumber(), 
+                                                SimTK::Vec3(SimTK::NaN) };
             int m{0};
             for(auto it = marker_pts->Begin();  it != marker_pts->End(); ++it) {
                 auto pt = *it;
-                row[m++] = SimTK::Vec3{pt->GetValues().coeff(f, 0),
-                                       pt->GetValues().coeff(f, 1),
-                                       pt->GetValues().coeff(f, 2)};
+                // BTK reads empty values as zero, but sets a "residual" value
+                // to -1 and it is how it knows to export these values as 
+                // blank, instead of 0,  when exporting to .trc
+                // Read in value if it is not zero or residual is not -1
+                if (!pt->GetValues().row(f).isZero() ||    //not precisely zero
```

Comment:
I'm still missing why the first condition is necessary. Wouldn't `if (pt->GetResiduals().coeff(f) != -1)` be sufficient? Does BTK set the "residual" flag only in some cases, or is the "residual" flag sometimes wrong? If the data from BTK is untrustworthy, then perhaps there should even be a check here that warns the user if the value and residual are contradictory (i.e., if `!pt->GetValues().row(f).isZero()` yet `pt->GetResiduals().coeff(f) == -1`)?

Reply:
The BTK code explicitly interprets the 0 and the residual to -1 as blank entries. I don't understand why I should not ONLY check for this condition to replace with a NaN. You may be right that the residual in only set to -1 when no value is encountered, but I do not know that for sure. I am 100% certain about the case I am testing for.

Reply:
> I don't understand why I should not ONLY check for this condition to replace with a NaN.

I'm not a BTK expert so I'm going by the comment, which seems to imply that the signal is simply whether `residual == -1`. The first half of the condition will accept nonzero values *regardless* of the state of the `residual` flag. I don't know whether nonzero values should be trusted if `residual == -1`.

> I am 100% certain about the case I am testing for.

Ok.